cli/commands/catalog/module/repo.go

```diff
@@ -181,18 +177,13 @@ func (repo *Repo) clone(ctx context.Context) error {
 	}
 	repo.cloneUrl = sourceUrl.String()
 
-	log.Infof("Cloning repository %q to temprory directory %q", repo.cloneUrl, tempDir)
-
-	// if no repo directory is specified, `go-getter` returns the error "git exited with 128: fatal: not a git repository (or any of the parent directories"
-	if !strings.Contains(sourceUrl.RequestURI(), "//") {
-		sourceUrl.Path += "//."
-	}
+	log.Infof("Cloning repository %q to temprory directory %q", repo.cloneUrl, repo.tempDir)
 
-	if err := getter.GetAny(tempDir, strings.Trim(sourceUrl.String(), "/"), getter.WithContext(ctx)); err != nil {
+	if err := getter.Get(repo.tempDir, strings.Trim(sourceUrl.String(), "/"), getter.WithContext(ctx)); err != nil {
```

Comment:
Two questions:

1. Is this method smart enough to update an existing repo if it's already checked out there?
2. Does this method do a full `git clone`? Or is there some way we can do a "lighter weight" one that _only_ checks out `main` and not all the other branches and full history?

Reply:
> Two questions:
> 
> 1. Is this method smart enough to update an existing repo if it's already checked out there?

Yes, It is.

https://github.com/hashicorp/go-getter/blob/975961f5f06346ccc282cd0d9aa16e160d26f9e3/get_git.go#L122-L130

> 2. Does this method do a full `git clone`? Or is there some way we can do a "lighter weight" one that _only_ checks out `main` and not all the other branches and full history?

This method only checks out the default branch.

https://github.com/hashicorp/go-getter/blob/975961f5f06346ccc282cd0d9aa16e160d26f9e3/get_git.go#L192-L204

Reply:
Sweeeeet!